library/src/main/java/com/novoda/viewpageradapter/ViewPagerAdapter.java

```diff
@@ -8,10 +8,13 @@
 import android.view.ViewGroup;
 
 import java.util.HashMap;
+import java.util.Locale;
 import java.util.Map;
 
 public abstract class ViewPagerAdapter<V extends View> extends PagerAdapter {
 
+    private static final String ILLEGAL_STATE_ID_ON_VIEW = "Found existing ID on view: %s. " + ViewPagerAdapter.class.getSimpleName() + " is responsible for setting/restoring page view IDs.";
```

Comment:
Maybe we can improve this: `The view created for position %d has an ID. Page views cannot have an ID because the adapter needs to control that to deal with the state saving and restoring. Make sure your inflated views have an ID of View.NO_ID`?
Not a blocker.